src/ImageSharp/Metadata/Profiles/Exif/Values/ExifNumberArray.cs

```diff
@@ -36,6 +38,54 @@ public override ExifDataType DataType
             }
         }
 
+        public override bool TrySetValue(object value)
+        {
+            if (base.TrySetValue(value))
+            {
+                return true;
+            }
+
+            switch (value)
+            {
+                case int val:
+                    return this.SetSingle(val, v => (Number)v);
+                case uint val:
+                    return this.SetSingle(val, v => (Number)v);
+                case short val:
+                    return this.SetSingle(val, v => (Number)v);
+                case ushort val:
+                    return this.SetSingle(val, v => (Number)v);
+                case int[] array:
+                    return this.SetArray(array, v => (Number)v);
+                case uint[] array:
+                    return this.SetArray(array, v => (Number)v);
+                case short[] array:
+                    return this.SetArray(array, v => (Number)v);
+                case ushort[] array:
+                    return this.SetArray(array, v => (Number)v);
+            }
+
+            return false;
+        }
+
         public override IExifValue DeepClone() => new ExifNumberArray(this);
+
+        private bool SetSingle<T>(T value, Func<T, Number> converter)
```

Comment:
This looks like we do a cast for every call. Can we remove the `Func`?

Reply:
Removed `Func` with a67bdc22